build/dependencies.yaml

```diff
@@ -101,30 +101,36 @@ dependencies:
 
   # Golang
   - name: "golang: upstream version"
-    version: 1.16.7
+    version: 1.17rc2
     refPaths:
-    - path: build/build-image/cross/VERSION
+    # TODO(releng): Uncomment once go1.17 is released
+    #- path: build/build-image/cross/VERSION
     - path: cluster/addons/fluentd-elasticsearch/es-image/Dockerfile
       match: 'FROM golang\:\d+.\d+(alpha|beta|rc)?\.?(\d+)?'
     - path: test/images/Makefile
       match: GOLANG_VERSION=\d+.\d+(alpha|beta|rc)?\.?\d+
     - path: staging/publishing/rules.yaml
       match: 'default-go-version\: \d+.\d+(alpha|beta|rc)?\.?(\d+)?'
 
+  # TODO(releng): Uncomment once go1.17 is released
+  - name: "golang: minimum supported version"
+    version: 1.16.0
+    refPaths:
+    - path: hack/lib/golang.sh
+      match: minimum_go_version=go([0-9]+\.[0-9]+)
```

Comment:
hmm... this is going to cause issues with people running gofmt and update-vendor and getting wildly different results locally with go 1.16.x